File: app/src/main/java/es/ulpgc/tfm/ecocsgo/model/Grenade.java
package es.ulpgc.tfm.ecocsgo.model;

import android.os.Parcel;
import android.os.Parcelable;

public class Grenade extends EquipmentJava implements Parcelable {
    public Grenade(String key){
        super("grenades", key);
        this.acceptedCategories = new EquipmentCategory[]{
                EquipmentCategory.GRENADE
        };
        getData();
    }

    protected Grenade(Parcel in) {
        super(in);
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        super.writeToParcel(dest, flags);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<Grenade> CREATOR = new Creator<Grenade>() {
        @Override
        public Grenade createFromParcel(Parcel in) {
            return new Grenade(in);
        }

        @Override
        public Grenade[] newArray(int size) {
            return new Grenade[size];
        }
    };
}

File: app/src/main/java/es/ulpgc/tfm/ecocsgo/model/Grenade.kt
package es.ulpgc.tfm.ecocsgo.model

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize

@Parcelize
data class Grenade(
    override var name: String, override var team: EquipmentTeamEnum,
    override var numeration: EquipmentNumeration, override var cost: Int
) :
    Equipment(name, team, numeration, cost), Parcelable